pkg/apis/core/validation/utils_test.go

```diff
@@ -91,4 +93,105 @@ var _ = Describe("Utils tests", func() {
+			capabilities core.Capabilities = map[string]string{
+				"architecture":   "amd64,arm64",
+				"hypervisorType": "gen1,gen2,gen3",
+			}
+		)
+		It("should accept a capabilitiesDefinition with architecture as capability", func() {
+			errorList := ValidateCapabilitiesDefinition(capabilities, dummyPath)
+			Expect(errorList).To(BeEmpty())
+
+		})
+
+		DescribeTable("should reject invalid capabilitiesDefinition",
+			func(capabilities core.Capabilities, expectedError []gomegatypes.GomegaMatcher) {
+				errorList := ValidateCapabilitiesDefinition(capabilities, dummyPath)
+				Expect(errorList).To(ConsistOf(expectedError))
+			},
+			Entry("reject empty capability name", core.Capabilities{"architecture": "amd64", "": "gen1,gen2,gen3"}, []gomegatypes.GomegaMatcher{
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeInvalid),
+					"Field": Equal(dummyPath.String()),
+				})),
+			}),
+			Entry("empty capability values", core.Capabilities{"architecture": "amd64", "hypervisorType": ""}, []gomegatypes.GomegaMatcher{
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeRequired),
+					"Field": Equal(dummyPath.Child("hypervisorType").String()),
+				})),
+			}),
+			Entry("missing architecture capability", core.Capabilities{"hypervisorType": "gen1,gen2,gen3"}, []gomegatypes.GomegaMatcher{
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeRequired),
+					"Field": Equal(dummyPath.Child("architecture").String()),
+				})),
+			}),
+		)
+
+		It("should unmarshal correct CapabilitiesSet", func() {
+			var CapabilitiesSet = []v1.JSON{
+				{Raw: []byte(`{"architecture":"amd64","hypervisorType":"gen2"}`)},
+				{Raw: []byte(`{"architecture":"amd64","hypervisorType":"gen2"}`)},
+				{Raw: []byte(`{"architecture":"arm64","hypervisorType":"gen1"}`)},
+				{Raw: []byte(`{"architecture":"arm64","hypervisorType":"gen2,gen3"}`)},
+			}
+			capabilitiesSet, err := UnmarshalCapabilitiesSet(CapabilitiesSet, dummyPath)
+
+			Expect(err).To(BeNil())
+			Expect(capabilitiesSet).To(HaveLen(4))
+			Expect(capabilitiesSet[0]).To(Equal(core.Capabilities{"architecture": "amd64", "hypervisorType": "gen2"}))
+		})
+
+		It("should error on unparsable CapabilitiesSet", func() {
+			var CapabilitiesSet = []v1.JSON{
+				{Raw: []byte(`{"architecture":"amd64","hypervisorType":"gen2"}`)},
+				{Raw: []byte(`{"architecture":"amd64","hype....🆘`)},
+			}
+
+			capabilitiesSet, err := UnmarshalCapabilitiesSet(CapabilitiesSet, dummyPath)
+
+			Expect(err).To(ConsistOf([]gomegatypes.GomegaMatcher{
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeInvalid),
+					"Field": Equal("dummy[1]"),
+				})),
+			}))
+			Expect(capabilitiesSet[1]).To(BeEmpty())
+		})
+
+		It("should sanitize capability values on parsing", func() {
+			var capabilities2 core.Capabilities = map[string]string{
+				"architecture":   "  amd64 ,arm64 ,amd32, , 'asdas   '    , I look weird  ",
```

Comment:
Imho the `'asdas ' ` case would only make sense, it my value (in this case an `architecture`) may itself contain commas that should be escaped (example: `asdas,gen.2`).
Is there a reference in the Gardener or Kubernetes context, where a similar sanitation is applied?